File: fixtures/scale/r/medium/funcs/fn_hypot.r
# Euclidean distance from the origin.

calc_hypot <- function(x, y) {
  stopifnot(is.numeric(x), is.numeric(y))
  sqrt(x * x + y * y)
}

hypot_table <- function(xs, ys) {
  mapply(calc_hypot, xs, ys)
}

File: fixtures/scale/fortran/medium/src/functions/fn_hypot.f90
module fn_hypot
  ! Euclidean distance from the origin.
  implicit none

contains

  function calc_hypot(a, b) result(y)
    real(kind=8), intent(in) :: a, b
    real(kind=8) :: y
    y = sqrt(a * a + b * b)
  end function calc_hypot

end module fn_hypot
